### DISCUSSION TITLE
Calculation of the offset of the phase

### DISCUSSION THREAD

**Discussion by mirichard-S** [carnisj/bcdi#308] [Q&A] [answered]
Hi Jérôme,

Can you remind me how the offset of the phase is calculated?
Is the phase equal to zero at the voxel corresponding to the center of mass of the reconstructed modulus?
Or is the phase offset equal to the mean of the phase inside the support?

Best,

**Comment by carnisj** [comment, marked as answer]
In `postprocessing.process_scan`:

![image](https://user-images.githubusercontent.com/44344327/188940878-a4146d21-cdef-4402-abc2-ad8aba76250d.png)

The parameters are already described in the example config file and docstrings:

`offset_method`: 'mean' or 'com' (center of mass). This phase value is subtracted to the phase
`phase_offset_origin`: pixel coordinates or None. If None, the phase origin is determined by `phase_offset`
`phase_offset`: use this to offset manually the origin of the phase to the desired value

**Comment by carnisj** [comment]
I will clean up a bit the function `remove_offset` and its docstring, but in principle one should be able to get the desired behavior with the existing parameters, correct?
